config/vufind/geofeatures.ini

```diff
@@ -0,0 +1,48 @@
+; Geographic feature configuration settings
+;
+; Currently all but the basemap configurations for the geographic feature
+; functionality are located in config.ini (for Map Tab display) and in
+; searches.ini (for Map Selection).
+
+[Basemaps]
```

Comment:
It seems that the configuration has evolved a little differently than I had expected -- which is not necessarily a bad thing. But just to clarify what I had expected...

I thought we'd have something like this:

<pre>
[Basemaps]
basemap_url[osm-intl] = https://maps.wikimedia.org/osm-intl/{z}/{x}/{y}.png
basemap_attribution[osm-intl] = "<a href=""https://wikimediafoundation.org/wiki/Maps_Terms_of_Use"">Wikimedia</a> | © <a href=""https://www.openstreetmap.org/copyright"">OpenStreetMap</a>"

basemap_url[stamen-toner] = http://tile.stamen.com/toner/{z}/{x}/{y}.png
basemap_attribution[stamen-toner] = "Map tiles by <a href=""http://stamen.com"">Stamen Design</a>, under <a href=""http://creativecommons.org/licenses/by/3.0"">CC BY 3.0</a>. Data by <a href=""http://openstreetmap.org"">OpenStreetMap</a>, under <a href=""http://www.openstreetmap.org/copyright"">ODbL</a>."

...etc.
</pre>

And then I assumed that some other setting could be set to osm-intl or stamen-toner, which would control which array elements from the [Basemaps] section would be loaded.

That's useful if you want to offer something like user-selectable basemaps, since it lets you define multiple settings at once. But if there's only going to be one setting at a time, then having a list of options and making the user simply uncomment one set is certainly more direct and simpler.

Does that make any sense?

Reply:
I did think about that, but thought the simpler approach would be easier for the user to specify a different basemap url and attribution by simply adding those values into the basemap_url = and basemap_attribution = settings. I know it's not the original direction, but I thought this would be simpler and more straightforward while allowing for a greater amount of flexibility for the long term. I saw this more as a system configuration setting rather than as a user-selectable option (like a map layer) on the map interface. Do you see any downsides to this approach?

Reply:
The only downside I see is the one I mentioned: that you can't define multiple basemaps for selection by the user using this approach... but if you never see a need to do that, it's not a problem... and honestly, even if that need arises, we might be able to add support in a backward-compatible way by doing an `is_array` check on the setting... so this is probably fine as it is. I just wanted to double-check!